Gumroad import improvements

oxi7589 opened the pull request:
This fixes problems that can arise during import of a Gumroad library where two different artists share the same display name, and allows importing purchases from stores that have been closed. (Products from closed stores don't have *any* data about the parent creator, so pre-filtering by creator is the only option to associate those with the appropriate creator.)

A fallback for obtaining the Gumroad artist's name is also implemented, since the normal way doesn't work for them (storefront returns 404): when the Gumroad importer finishes, it passes an extra parameter to the `index_artists()` containing a mapping of ID to a display name which can be used to properly name 404'ed stores. Calls to `index_artists()` from other importer threads simply skip these artists without raising exceptions.

BassOfBass commented:
> This fixes problems that can arise during import of a Gumroad library where two different artists share the same display name, and allows importing purchases from stores that have been closed. (Products from closed stores don't have _any_ data about the parent creator, so pre-filtering by creator is the only option to associate those with the appropriate creator.)
> 
> A fallback for obtaining the Gumroad artist's name is also implemented, since the normal way doesn't work for them (storefront returns 404): when the Gumroad importer finishes, it passes an extra parameter to the `index_artists()` containing a mapping of ID to a display name which can be used to properly name 404'ed stores. Before that calls to `index_artists()` from other importer threads simply skip these artists without raising exceptions.

This info should be added within the code as separate comment lines before the related blocks or as docstrings for named functions. Also provide type annotations for any new functions you introduce. Don't need to be super anal about them, `Dict` will suffice for any `dict` etc.

oxi7589 commented:
I've updated the code to include better inline documentation. 

> Dict will suffice for any dict etc.

Got it. I've specifically chosen to use fully-qualified `List[str]` in the function I've introduced to explicitly note that the numeric creator ID contained in this list is also a `str`.
